### PR TITLE
Added API server load balancers HA requirement for production use

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://github.com/openshift/openshift-docs/pull/9585

@openshift/team-documentation PTAL

### CODE REVIEW

File: architecture/infrastructure_components/kubernetes_infrastructure.adoc [hunk 1]
```diff
@@ -147,6 +147,18 @@ maintain a uniform odd number of hosts when collocating master services and
...
 
+[NOTE]
+====
+In production {product-title} clusters, you must maintain high availability
+of the API Server load balancer. If the API Server load balancer is not
+available, nodes cannot report their status, all their pods are marked dead,
+and the pods' endpoints are removed from the service.
+
+You must separately configure HA for the API Server load balancer. To configure
+HA for it, you can run multiple HAProxy load balancers and use Keepalived to
```
Comment (bfallonf) [inline architecture/infrastructure_components/kubernetes_infrastructure.adoc]: Maybe s/To configure HA for it,/To do this,

We use this in other parts of the docs.

But also, is there somewhere in the docs that explains how to to this? Maybe link out to there?

Comment (kalexand-rh) [reply architecture/infrastructure_components/kubernetes_infrastructure.adoc]: Thanks, @bfallonf! The source bug says that it's not something that you do in the product, that it's completely separate. There's nothing in the docs for it, and while this one method of configuring 
How's this?
 "In addition to configuring HA for {product-title}, you must separately configure HA for the API Server load balancer. One HA option is to run multiple HAProxy load balancers and use Keepalived to provide a floating virtual IP address for them."

Comment (bfallonf) [reply architecture/infrastructure_components/kubernetes_infrastructure.adoc]: @kalexand-rh Hmm sure. I think I have a BZ about creating docs for this maybe... +1 to your suggestion.

Comment (kalexand-rh) [reply architecture/infrastructure_components/kubernetes_infrastructure.adoc]: Well, if you have a BZ about it, then I guess we do need to be opinionated. :)
I'll update this change and merge. Thanks @bfallonf!

**Comment by kalexand-rh** [comment]
Fixing the merge conflicts on these cherry-picks made me question if the ifdef statements for the subsection this note's in are right. I'm getting more information and will make corrections as needed.

**Comment by kalexand-rh** [comment]
I reverted this change on https://github.com/openshift/openshift-docs/pull/10341, per an email from @acomabon, and closed all the PRs.

**Pull request merged**